api/src/main/java/org/opennms/integration/api/v1/topology/TopologyEdgeConsumer.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.integration.api.v1.topology;
+
+import org.opennms.integration.api.v1.annotations.Exposable;
+import org.opennms.integration.api.v1.model.TopologyEdge;
+
+/**
+ * Implementations of this interface will receive {@link TopologyEdge edges} from OpenNMS.
+ *
+ * @author mbrooks
+ * @since 1.0.0
+ */
+@Exposable
+public interface TopologyEdgeConsumer {
+    /**
+     * @param topologyEdge the edge being added or updated
+     */
+    void onEdgeAddedOrUpdated(TopologyEdge topologyEdge);
+
+    /**
+     * @param topologyEdge the edge being deleted
+     */
+    void onEdgeDeleted(TopologyEdge topologyEdge);
+
+    /**
+     * Provides the list of protocols that this consumer supports. Only edges corresponding to the given protocols will
+     * be sent via {@link #onEdgeAddedOrUpdated(TopologyEdge)} and {@link #onEdgeDeleted(TopologyEdge)}.
+     *
+     * @return the comma separated list of protocols this consumer supports
+     */
+    String getProtocols();
```

Comment:
Can we make it possible to register for all protocols without having to specify them?